torch/_dynamo/variables/tensor.py

```diff
@@ -217,15 +217,25 @@ def try_generic_attr_handling():
 
         return result
 
+    def has_unpack_var_sequence(self, tx):
+        return (self.size is not None and len(self.size) > 0) or (
+            self.size is None and config.dynamic_shapes
+        )
+
     def unpack_var_sequence(self, tx, idxes=None):
         from .builder import wrap_fx_proxy
 
+        options = VariableTracker.propagate(self)
         if idxes is None:
             if self.size:
-                idxes = range(self.size[0])
+                length = self.size[0]
             else:
-                return super(TensorVariable, self).unpack_var_sequence(tx)
-        options = VariableTracker.propagate(self)
+                dyn_length = self.call_method(
+                    tx, "size", [ConstantVariable(0, **options)], {}
```

Comment:
There isn't really point in propagating the self options into the fresh ConstantVariable. ConstantVariable(0) is fine.